### ISSUE TITLE
yolov8 onnx static quantized

### ISSUE THREAD

**Issue by venxzw** [ultralytics/ultralytics#16468]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I use the dynamic quantization provided by ONNXRuntime to quantize the ONNX model trained, I found that some detection regions have multiple overlapping detection boxes during prediction. Is there any reason for this? Is it because of my incorrect exclusion of nodes?

### Additional

`import numpy as npfrom onnxruntime.quantization import CalibrationDataReader, quantize_static, QuantType, QuantFormat    # Class for Callibration Data readingclass ImageCalibrationDataReader(CalibrationDataReader):    def __init__(self, image_paths):        self.image_paths = image_paths        self.idx = 0        self.input_name = "images"
    def preprocess(self, frame):        # Same preprocessing that you do before feeding it to the model        frame = cv2.imread(frame)        X = cv2.resize(frame, (640, 640))        image_data = np.array(X).astype(np.float32) / 255.0  # Normalize to [0, 1] range        image_data = np.transpose(image_data, (2, 0, 1))  # (H, W, C) -> (C, H, W)        image_data = np.expand_dims(image_data, axis=0)  # Add batch dimension        return image_data
    def get_next(self):        # method to iterate through the data set        if self.idx >= len(self.image_paths):            return None
        image_path = self.image_paths[self.idx]        input_data = self.preprocess(image_path)        self.idx += 1        return {self.input_name: input_data}
# Assuming you have a list of image paths for calibrationcalibration_image_paths = ['test.jpg'] # you can add more of the image paths
# Create an instance of the ImageCalibrationDataReadercalibration_data_reader = ImageCalibrationDataReader(calibration_image_paths)
quantize_static('preprocessed.onnx', "static_quantized.onnx",                weight_type=QuantType.QInt8,                activation_type=QuantType.QUInt8,                calibration_data_reader=calibration_data_reader,                quant_format=QuantFormat.QDQ,                nodes_to_exclude=['/model.22/Concat_3', '/model.22/Split', '/model.22/Sigmoid'                                 '/model.22/dfl/Reshape', '/model.22/dfl/Transpose', '/model.22/dfl/Softmax',                                  '/model.22/dfl/conv/Conv', '/model.22/dfl/Reshape_1', '/model.22/Slice_1',                                 '/model.22/Slice', '/model.22/Add_1', '/model.22/Sub', '/model.22/Div_1',                                  '/model.22/Concat_4', '/model.22/Mul_2', '/model.22/Concat_5'],                per_channel=False,                reduce_range=True,)`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @venxzw, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for valuable insights. For quantization issues, you may find helpful guidance in the [EdgeAI Quantization Guide](https://docs.ultralytics.com/guides/edgeai-quantization/).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in debugging the issue. If your concern involves node exclusions, adding more details about your setup and configuration can be particularly useful.

Join the Ultralytics community where it suits you best. For real-time discussions, visit our [Discord](https://ultralytics.com/discord) 🎧. For in-depth problem-solving, check out [Discourse](https://community.ultralytics.com), or engage with the community on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure you're benefiting from the latest features and fixes, upgrade the `ultralytics` package and verify your issue in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in various verified environments with preinstalled dependencies. Try it free in [Colab](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) or check out other options like [Kaggle](https://www.kaggle.com/ultralytics/yolov8) or [Docker](https://hub.docker.com/r/ultralytics/ultralytics).

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify correct operation on macOS, Windows, and Ubuntu every 24 hours.

This is an automated response, and an Ultralytics engineer will assist you soon. 😊

**Comment by Skillnoob** [comment]
Can you share the code you use for predicting?

**Comment by Christian-lyc** [comment]
Thank you for sharing the code. It was really helpful. Thank you

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by venxzw** [comment]
I have already solved the problem. It was indeed a problem with the node exclusion.

**Comment by Skillnoob** [comment]
@venxzw Nice to hear. It would be great if you could share the full solution for anyone who runs into this issue in the future
